auth.py

```diff
@@ -15,7 +15,11 @@
 cookie_name     = config ['session'] ['cookie_name']
 session_length  = config ['session'] ['session_length'] * 60
 
-env = os.getenv ('HEROKU_APP_NAME')
+def is_e2e_test_request (email):
```

Comment:
How about a slightly different approach that doesn't rely on magic email addresses?

What if we put a special header:

```
X-Testing: 1
```

In the request we send from the test?

Then in the `app.py` we can do:

```python
if os.getenv('IS_PRODUCTION'):
  @app.before_request
  def reject_tests():
    if is_testing_request():
      # This probably needs to return a response instead of throwing, but you get the idea :)
      raise RuntimeError('No testing in production!')
```

And if we want to skip certain behavior like sending email for tests:

```python
def is_testing_request():
  return request.headers.get('X-Testing')
```

Reply:
This is a great idea! I will use a header instead.

Reply:
I guess this then is the only change I am awaiting before merge?